### PR TITLE
Fixes #24901 - Validate interval with cron expression

### PR DESCRIPTION
**Pull request by sjha4**
- Cron expression should be unset if interval is not custom cron
 - Adding a validation to make sure cron expression is not set when interval is not custom cron.

### CODE REVIEW

**Comment by omkarkhatavkar** [comment]
Looks good @sjha4 

I ran this with the nailgun, tests are getting passed on API level.
```

>>> for i in ('hourly', 'daily', 'weekly'):
...     try:
...         sync_plan = SyncPlan(organization=org, name='test5', interval=i, enabled='True', sync_date=datetime.now(),
...                          cron_expression='* * * * *').create()
...     except Exception as ex:
...         print(ex)
...         
... 
WARNING:nailgun.client:Received HTTP 422 response: {"displayMessage":"Validation failed: Custom cron expression only needs to be set for interval value of custom cron","errors":{"base":["Custom cron expression only needs to be set for interval value of custom cron"]}}
422 Client Error: Unprocessable Entity for url: https://centos7-devel.okhatavk.example.com/katello/api/v2/organizations/47/sync_plans
WARNING:nailgun.client:Received HTTP 422 response: {"displayMessage":"Validation failed: Custom cron expression only needs to be set for interval value of custom cron","errors":{"base":["Custom cron expression only needs to be set for interval value of custom cron"]}}
422 Client Error: Unprocessable Entity for url: https://centos7-devel.okhatavk.example.com/katello/api/v2/organizations/47/sync_plans
WARNING:nailgun.client:Received HTTP 422 response: {"displayMessage":"Validation failed: Custom cron expression only needs to be set for interval value of custom cron","errors":{"base":["Custom cron expression only needs to be set for interval value of custom cron"]}}
422 Client Error: Unprocessable Entity for url: https://centos7-devel.okhatavk.example.com/katello/api/v2/organizations/47/sync_plans

```

File: app/models/katello/sync_plan.rb [hunk 1]
```diff
@@ -163,5 +169,9 @@ def valid_cron_logic?(cron_logic)
     rescue ArgumentError => _
       false
     end
+
+    def cron_status_mismatch?
+      self.interval != 'custom cron' && !(self.cron_expression.nil? || self.cron_expression.eql?(''))
```
Comment (jlsherrill) [inline app/models/katello/sync_plan.rb]: can you use the constant 'CUSTOM_CRON' here?

Comment (jlsherrill) [reply app/models/katello/sync_plan.rb]: and everywhere else in the file that references 'custom cron'

**Comment by sjha4** [comment]
Thanks @omkarkhatavkar , @jlsherrill !! :+1:

**Pull request merged**